### PR TITLE
Update to PHPStan v0.11

### PR DESCRIPTION
**Pull request by carusogabriel**
Although we're not in a higher level of PHPStan, let at least keep with the last version of it :)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "require-dev": {
         "ext-pdo": "*",
         "doctrine/coding-standard": "^6.0",
-        "phpstan/phpstan-shim": "^0.9.2",
+        "phpstan/phpstan": "^0.11",
```
Comment (greg0ire) [inline composer.json]: I don't understand how it was working before… shouldn't it have been `vendor/bin/phpstan.phar` in `.travis.yml`? Also, why this change?

Comment (carusogabriel) [reply composer.json]: `phpstan-shim` is the binary version and I think we're using it to avoid conflicts. I've changed it to align the way we do in other repositores.

File: lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php [hunk 2]
```diff
@@ -93,7 +93,6 @@ class BasicEntityPersister implements EntityPersister
 {
     /** @var string[] */
     private static $comparisonMap = [
-        Comparison::EQ          => '= %s',
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php]: Why is this being removed? Eq/Is being the same?

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php (hunk 2 above)
Comment (carusogabriel) [reply lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php]: Yeap, here's the PHPStan error:

```
 ------ ----------------------------------------------------------------------------------------------
  Line   Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php
 ------ ----------------------------------------------------------------------------------------------
  96     Array has 2 duplicate keys with value '=' (\Doctrine\Common\Collections\Expr\Comparison::EQ,
         \Doctrine\Common\Collections\Expr\Comparison::IS).
 ------ ----------------------------------------------------------------------------------------------
```

**Review by Ocramius — APPROVED** [review]
